server/lorax_server/utils/sources/__init__.py

```diff
@@ -7,6 +11,30 @@
 HUB = "hub"
 S3 = "s3"
 LOCAL = "local"
+PBASE = "pbase"
+
+PREDIBASE_MODEL_URL_ENDPOINT = "/v1/models/version/name/{}"
+PREDIBASE_MODEL_VERSION_URL_ENDPOINT = "/v1/models/version/name/{}?version={}"
+PREDIBASE_GATEWAY_ENDPOINT = os.getenv("PREDIBASE_GATEWAY_ENDPOINT", "https://api.predibase.com")
+
+
+@lru_cache(maxsize=256)
+def map_pbase_model_id_to_s3(model_id: str, api_token: str) -> str:
```

Comment:
API token is optional, right? We should raise a useful error if the user didn't provide it.